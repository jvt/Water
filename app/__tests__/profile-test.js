import {
    TextInput,
    Text,
    Image
} from 'react-native';
import React from 'react';
import Profile from '../profile'
import Button from 'apsl-react-native-button';
import SegmentedControlTab from 'react-native-segmented-control-tab';
import BusyIndicator from 'react-native-busy-indicator';
import Renderer from 'react-test-renderer';

describe('Profile', () => {
	test('renders correctly', () => {
		const profile = Renderer.create(
            <Profile />
        ).toJSON();
        expect(profile).toMatchSnapshot();
	})
})