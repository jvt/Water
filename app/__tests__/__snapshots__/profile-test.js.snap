exports[`Profile HomeAddressField 1`] = `
<TextInput
  HomeAddress={true}
  Style={Object {}} />
`;

exports[`Profile HomeAddressText 1`] = `
<Text
  Style={Object {}}
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail">
  Home Address:
</Text>
`;

exports[`Profile MyAccountText 1`] = `
<Text
  Style={Object {}}
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail">
  My Account
</Text>
`;

exports[`Profile ReturnButton 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "marginBottom": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "fontSize": 18,
          "textAlign": "center",
        },
        undefined,
      ]
    }>
    Return
  </Text>
</View>
`;

exports[`Profile RoleTabs 1`] = `
<View
  removeClippedSubviews={true}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flexDirection": "row",
      },
      Object {},
    ]
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0076FF",
        "borderBottomLeftRadius": 5,
        "borderColor": "#0076FF",
        "borderTopLeftRadius": 5,
        "borderWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          undefined,
          Object {
            "color": "white",
          },
          undefined,
        ]
      }>
      User
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#0076FF",
        "borderWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          undefined,
          Object {},
          Object {},
        ]
      }>
      Worker
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#0076FF",
        "borderWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          undefined,
          Object {},
          Object {},
        ]
      }>
      Manager
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomRightRadius": 5,
        "borderColor": "#0076FF",
        "borderTopRightRadius": 5,
        "borderWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          undefined,
          Object {},
          Object {},
        ]
      }>
      Admin
    </Text>
  </View>
</View>
`;

exports[`Profile RoleText 1`] = `
<Text
  Style={Object {}}
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail">
  Role:
</Text>
`;

exports[`Profile SaveButton 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "marginBottom": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "fontSize": 18,
          "textAlign": "center",
        },
        undefined,
      ]
    }>
    Save
  </Text>
</View>
`;

exports[`Profile TitleField 1`] = `
<TextInput
  Style={Object {}}
  Title={true} />
`;

exports[`Profile TitleText 1`] = `
<Text
  Style={Object {}}
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail">
  Title:
</Text>
`;

exports[`Profile UsernameField 1`] = `
<TextInput
  Style={Object {}}
  Username={true} />
`;

exports[`Profile UsernameText 1`] = `
<Text
  Style={Object {}}
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail">
  Username:
</Text>
`;

exports[`Profile renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flex": 1,
      },
      Object {
        "backgroundColor": "rgba(240, 240, 240, 1)",
      },
    ]
  }>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "paddingBottom": 50,
            "paddingTop": 50,
          }
        }>
        <View
          elevation={1}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 2,
              "margin": 10,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 10,
              "padding": 15,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 2,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 20,
                "fontWeight": "200",
                "textAlign": "center",
              }
            }>
            My Account
          </Text>
        </View>
        <View
          elevation={1}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 2,
              "margin": 10,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 10,
              "padding": 15,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 2,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 20,
                "fontWeight": "200",
                "textAlign": "left",
              }
            }>
            Username:
          </Text>
          <TextInput
            onChangeText={[Function]}
            style={
              Object {
                "height": 40,
              }
            }
            value={null} />
        </View>
        <View
          elevation={1}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 2,
              "margin": 10,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 10,
              "padding": 15,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 2,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 20,
                "fontWeight": "200",
                "textAlign": "left",
              }
            }>
            Home Address:
          </Text>
          <TextInput
            onChangeText={[Function]}
            style={
              Object {
                "height": 40,
              }
            }
            value={null} />
        </View>
        <View
          elevation={1}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 2,
              "margin": 10,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 10,
              "padding": 15,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 2,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 20,
                "fontWeight": "200",
                "textAlign": "left",
              }
            }>
            Role:
          </Text>
          <View
            style={
              Object {
                "padding": 20,
                "paddingBottom": 0,
              }
            }>
            <View
              removeClippedSubviews={true}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "flexDirection": "row",
                  },
                  Object {},
                ]
              }>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#0076FF",
                    "borderBottomLeftRadius": 5,
                    "borderColor": "#0076FF",
                    "borderTopLeftRadius": 5,
                    "borderWidth": 1,
                    "flex": 1,
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingVertical": 5,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      undefined,
                      undefined,
                      Object {
                        "color": "white",
                      },
                      undefined,
                    ]
                  }>
                  User
                </Text>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderColor": "#0076FF",
                    "borderWidth": 1,
                    "flex": 1,
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingVertical": 5,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      undefined,
                      undefined,
                      Object {},
                      Object {},
                    ]
                  }>
                  Worker
                </Text>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderColor": "#0076FF",
                    "borderWidth": 1,
                    "flex": 1,
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingVertical": 5,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      undefined,
                      undefined,
                      Object {},
                      Object {},
                    ]
                  }>
                  Manager
                </Text>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderBottomRightRadius": 5,
                    "borderColor": "#0076FF",
                    "borderTopRightRadius": 5,
                    "borderWidth": 1,
                    "flex": 1,
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingVertical": 5,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      undefined,
                      undefined,
                      Object {},
                      Object {},
                    ]
                  }>
                  Admin
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          elevation={1}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 2,
              "margin": 10,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 10,
              "padding": 15,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 2,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 20,
                "fontWeight": "200",
                "textAlign": "left",
              }
            }>
            Title:
          </Text>
          <TextInput
            onChangeText={[Function]}
            style={
              Object {
                "height": 40,
              }
            }
            value={null} />
        </View>
        <View
          elevation={1}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 2,
              "margin": 10,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 10,
              "padding": 15,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 2,
            }
          }>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "rgba(63, 209, 127, 1)",
                "borderRadius": 8,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 44,
                "justifyContent": "center",
                "marginBottom": 10,
                "marginLeft": 10,
                "marginRight": 10,
                "marginTop": 10,
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "flex": 1,
                    "fontSize": 18,
                    "textAlign": "center",
                  },
                  Object {
                    "fontSize": 18,
                  },
                ]
              }>
              Save
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "rgba(66, 163, 221, 1)",
                "borderRadius": 8,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 44,
                "justifyContent": "center",
                "marginBottom": 10,
                "marginLeft": 10,
                "marginRight": 10,
                "marginTop": 10,
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "flex": 1,
                    "fontSize": 18,
                    "textAlign": "center",
                  },
                  Object {
                    "fontSize": 18,
                  },
                ]
              }>
              Return
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View />
</View>
`;
